spacy/util.py

```diff
@@ -954,23 +954,36 @@ def run_command(
         when you want to run the command more like a function.
     RETURNS (Optional[CompletedProcess]): The process object.
     """
-    if isinstance(command, str):
-        # On Windows, it's possible for the split here to be different from what
-        # will actually be used for execution, but it will *usually* be correct,
-        # and is only used in debugging output if a command failed.
-        cmd_list = shlex.split(command, posix=True)
-        cmd_str = command
+
+    # The actual command to execute
+    cmd_to_run = None
+    # these are used for debugging
+    cmd_str = None
+    tool = None
+
+    if is_windows:
+        # On Windows we can pass a list or string. In this case we'll just pass
+        # what we're given.
+        cmd_to_run = command
+        if isinstance(command, str):
+            cmd_str = command
+        else:  # list
+            # TODO list2cmdline is an intentionally undocumented internal Python
+            # function, check if we want to use this.
+            cmd_str = subprocess.list2cmdline(command)
```

Comment:
`subprocess.list2cmdline` has been in Python since v2 and been stable, but it's intentionally undocumented and not designed for end-user use. Because of that I think we should avoid using it, though if we do I'm not sure what to do for the `cmd_str` variable, which is used if an error occurs to show what command was run. I guess we could copy the [current implementation of `list2cmdline`](https://github.com/python/cpython/blob/8184f0fce3b734413e3d3a282f1425d3cb8507fd/Lib/subprocess.py#L574) internally.

Reply:
I think it's ok to use it. I assume our tests would catch significant changes - and if so, we can vendor the code. I see the counter-argument as well though.

Whatever we decide, we should drop the `TODO` before merging.